src/util/strconv.h

```diff
@@ -164,29 +168,32 @@ static size_t unicode_tolower(char *encoded, size_t in_len) {
   }
   RS_LOG_ASSERT_FMT(i == u_len, "i (%u) should be equal to u_len (%zd)", i, u_len);
   // Encode Unicode codepoints back to utf8 string
-  ssize_t reencoded_len = nu_bytenlen(u_buffer, u_len, nu_utf8_write);
-  if (reencoded_len > 0 && reencoded_len <= in_len) {
-    nu_writenstr(u_buffer, u_len, encoded, nu_utf8_write);
-  } else {
-    reencoded_len = 0;
-    // If the reencoded length is greater than in_len, we cannot write it back
-    // to the original buffer, so we return 0.
-    // It could happen if the original string contains characters that
-    // require more bytes to represent in lower case than the original string.
-    // For example, if the original string contains a character that is
-    // represented by 2 bytes in utf8, and the lower case version of that
-    // character is represented by 3 bytes, the reencoded length will be
-    // greater than in_len, and we cannot write it back to the original buffer.
-    // To support this, we need to allocate a new buffer, and it is considered
-    // as part of MOD-8799 which has not been implemented yet.
+  ssize_t reencoded_len = nu_bytenlen(u_buffer, i, nu_utf8_write);
+  if (reencoded_len > 0) {
```

Comment:
Add comment about no need for an else for negative return value: can return zero when there is nothing to lower - which is OK, or a positive value?

Reply:
On second thought can set `out_len` to `in_len` in the else

Reply:
OK. Done.